v2/index/mhindexsorted_test.go

```diff
@@ -46,6 +46,33 @@ func TestMultiWidthCodedIndex_MarshalUnmarshal(t *testing.T) {
 	requireContainsAll(t, umSubject, records)
 }
 
+func TestMultiWidthCodedIndex_StableIterate(t *testing.T) {
+	rng := rand.New(rand.NewSource(1414))
+	records := generateIndexRecords(t, multihash.SHA2_256, rng)
```

Comment:
Did this test fail before your fixes? I think the two sources of non-determinism were the maps for "multihash algorithm" and "digest length", so I think you should use two concatenated lists of records - so that each list is different on both axes.

Reply:
(Otherwise I think your test only ends up with one entry in each of the two layers of maps)